pkg/registry/core/serviceaccount/storage/token.go

```diff
@@ -103,6 +104,14 @@ func (r *TokenREST) Create(ctx context.Context, name string, obj runtime.Object,
 	}
 	svcacct := svcacctObj.(*api.ServiceAccount)
 
+	if len(req.UID) > 0 && req.UID != svcacct.UID {
+		if utilfeature.DefaultFeatureGate.Enabled(genericfeatures.TokenRequestServiceAccountUIDValidation) {
+			return nil, errors.NewConflict(schema.GroupResource{Group: gvk.Group, Resource: gvk.Kind}, name, fmt.Errorf("the UID in the token request (%s) does not match the UID of the service account (%s)", req.UID, svcacct.UID))
```

Comment:
Can you update the existing token request integration tests to exercise this case?

Reply:
Good call on requiring integration tests. I didn't realize we had them for this. While adding integration tests, I realized the UID gets stripped from the incoming request by the API server:

[https://github.com/kubernetes/kubernetes/blob/d269d10bfc5f989da1a8fa24f64812f8040a2c65/staging/src/k8s.io/apiserver/pkg/registry/rest/meta.go#L29-L36](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html)

I implemented `SubresourceObjectMetaPreserver` on the `TokenREST` struct to bypass this behavior and added tests that are now passing.

Reply:
A potential follow-up we can consider: we have integration tests for client-go credential plugins.  Maybe we should have them for kubelet credential plugins too?  We have some integration tests for the pod certs work so it could make sense here as well.

Reply:
Makes sense! I'll open a follow-up issue and look into how we can get the tests added.